examples/shrink_grow_layout.rs

```diff
@@ -0,0 +1,131 @@
+//! This example demonstrates how to use a [on_layout](kayak_core::WidgetProps::get_on_layout)
+//! event in widgets.
+//!
+//! The problem here is strictly contrived for example purposes.
+//! We use grow/shrink buttons to set the value of a `width` bound to an [crate::Background] element's width
+//! On change of layout we print current width of that element and update the text of Width label.
+use bevy::{
+    prelude::{App as BevyApp, AssetServer, Commands, Res, ResMut},
+    window::WindowDescriptor,
+    DefaultPlugins,
+};
+use kayak_core::{
+    styles::{LayoutType, Style, StyleProp, Units, Edge},
+    OnLayout,
+};
+use kayak_core::{Color, EventType, OnEvent};
+use kayak_render_macros::use_state;
+use kayak_ui::widgets::{App, Text, Window, Element};
+use kayak_ui::{
+    bevy::{BevyContext, BevyKayakUIPlugin, FontMapping, UICameraBundle},
+    widgets::Button,
+};
+use kayak_ui::{
+    core::{render, rsx, widget, Index},
+    widgets::Background,
+};
+
+/// This widget provides a theme to its children
+#[widget]
+fn GrowShrink() {
+    // This is width of background element we update via buttons
+    let (background_width, set_width, _) = use_state!(150.0);
+
+    let panel_style = Style {
+        layout_type: StyleProp::Value(LayoutType::Row),
+        width: StyleProp::Value(Units::Auto),
+        height: StyleProp::Value(Units::Pixels(50.0)),
+        offset: StyleProp::Value(Edge::all(Units::Pixels(10.0))),
+        ..Default::default()
+    };
+
+    // Grow/Shrink button styles
+    let button_styles = Style {
+        width: StyleProp::Value(Units::Pixels(100.0)),
+        height: StyleProp::Value(Units::Pixels(24.0)),
```

Comment:
Last change I think (padding was a bit off when running):

```suggestion
        height: StyleProp::Value(Units::Pixels(30.0)),
```